src/components/filePreview/filePreview.tsx

```diff
@@ -1,33 +1,93 @@
 import './filePreview.css'
+import '../../index.css'
 
 import Card from '@mui/material/Card'
+import IconButton from '@mui/material/IconButton'
+import Modal from '@mui/material/Modal'
 import { useTheme } from '@mui/material/styles'
+import Tooltip from '@mui/material/Tooltip'
 import Typography from '@mui/material/Typography'
-import React from 'react'
+import React, { useState } from 'react'
 
 import { shortenString } from '../helper'
+import Icon from '../icon/icon'
+import PDFViewer from '../pdfViewer/pdfViewer'
 import type { FileData } from '../types'
 
 export interface FilePreviewProps {
   file: FileData
 }
 
+const supportedViewers: {
+  [key: string]: React.ComponentType<{ url: string }>
+} = {
+  pdf: PDFViewer,
+  // Add more mappings as needed for different file types
+}
```

Comment:
Could this be a prop with some defaults so that its easy for configuring this at application level?